src/components/NameEditor.tsx

```diff
@@ -0,0 +1,62 @@
+import { EditOutlined } from "@ant-design/icons";
+import { Button, Form, Input } from "antd";
+import React, { useState } from "react";
+
+/**
+ * Allows the user to edit text inline by rendering
+ * the string as a button that converts the string 
+ * into a text field when clicked by the user.
+ */
+export const NameEditor = ({ name, setName }: {
+  name: string;
+  setName: (name: string) => void;
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [form] = Form.useForm();
+
+  const onCancel = () => {
+    form.resetFields();
+    setIsEditing(false);
+  };
+
+  const onFinish = (value) => {
+    setIsEditing(false);
+    setName(value.name);
+  };
+
+  return isEditing ? (
+    <Form
+      form={form}
+      name="formData"
+      layout="inline"
+      onFinish={onFinish}
+      style={{ justifyContent: "center" }}
+    >
+      <Form.Item
+        name="name"
+        initialValue={name}
+        rules={[{ required: true, message: "Name is required." }]}
+      >
+        <Input data-testid={`${name}-input`} />
+      </Form.Item>
+      <Form.Item>
+        <Button type="text" onClick={onCancel} data-testid={`${name}-cancel`}>
+          Cancel
+        </Button>
+        <Button type="ghost" htmlType="submit" data-testid={`${name}-save`}>
+          Save
+        </Button>
+      </Form.Item>
+    </Form>
+  ) : (
+    <Button
+      type="text"
+      size="large"
+      icon={<EditOutlined style={{ fontSize: "14px" }} />}
```

Comment:
I think you should just get rid of this icon. It looks weird and this edit functionality is definitely an edge case. I think it's fine if the user just has to hover over the button for now. Otherwise you could put a really small edit button inline inside the button.

Reply:
Your comment gave me an idea. I made it so the icon only shows on hover.

Reply:
Even better! 🎉